### ISSUE TITLE
hi_xiaowen-mdtc perfermance can't be reproduced

### ISSUE THREAD

**Issue by sugarcase** [wenet-e2e/wekws#69]
I followed the stages in "$root_path/examples/hi_xiaowen/s0/run.sh" , 
and try to reproduce network perfermance using config/mdtc.yaml, num_average=10, max_epoch=80,
but after training and check results(score.txt, stats.0.txt, stats.1.txt), the score, fa and recall seems very strange:
![image](https://user-images.githubusercontent.com/23147096/164368764-9c8b4a01-6496-4555-bee9-c042fec3c339.png)
![image](https://user-images.githubusercontent.com/23147096/164368880-56704231-b790-474b-98ce-98a1c845b550.png)
1. the highest score comes out at first 1/2 frame,  the keyword speech has not be spoken at that time.
2. most scores are lower than 0.5.
3. because we config two keywords, but most postive wav got valuable and high score at both two output points.
this is strange, when i change to ds_tcn config, the score seems correct.
![image](https://user-images.githubusercontent.com/23147096/164369863-1bc923ac-9d69-47cd-8cf3-59c7f454639f.png)

**Comment by robin1001** [comment]
The wakeup position is not accurate now, we are trying to figure it out.

**Comment by yangyyt** [comment]
> I followed the stages in "$root_path/examples/hi_xiaowen/s0/run.sh" , and try to reproduce network perfermance using config/mdtc.yaml, num_average=10, max_epoch=80, but after training and check results(score.txt, stats.0.txt, stats.1.txt), the score, fa and recall seems very strange: ![image](https://user-images.githubusercontent.com/23147096/164368764-9c8b4a01-6496-4555-bee9-c042fec3c339.png) ![image](https://user-images.githubusercontent.com/23147096/164368880-56704231-b790-474b-98ce-98a1c845b550.png)
> 
> 1. the highest score comes out at first 1/2 frame,  the keyword speech has not be spoken at that time.
> 2. most scores are lower than 0.5.
> 3. because we config two keywords, but most postive wav got valuable and high score at both two output points.
>    this is strange, when i change to ds_tcn config, the score seems correct.
>    ![image](https://user-images.githubusercontent.com/23147096/164369863-1bc923ac-9d69-47cd-8cf3-59c7f454639f.png)
but when I use thr ds_tcn config，the result is still not reproducible. eg. FAR=1, FRR=0.98。

**Comment by yangyyt** [comment]
but when I use thr ds_tcn config，the result is still not reproducible. eg. FAR=1, FRR=0.98

**Comment by yangyyt** [comment]
> I followed the stages in "$root_path/examples/hi_xiaowen/s0/run.sh" , and try to reproduce network perfermance using config/mdtc.yaml, num_average=10, max_epoch=80, but after training and check results(score.txt, stats.0.txt, stats.1.txt), the score, fa and recall seems very strange: ![image](https://user-images.githubusercontent.com/23147096/164368764-9c8b4a01-6496-4555-bee9-c042fec3c339.png) ![image](https://user-images.githubusercontent.com/23147096/164368880-56704231-b790-474b-98ce-98a1c845b550.png)
> 
> 1. the highest score comes out at first 1/2 frame,  the keyword speech has not be spoken at that time.
> 2. most scores are lower than 0.5.
> 3. because we config two keywords, but most postive wav got valuable and high score at both two output points.
>    this is strange, when i change to ds_tcn config, the score seems correct.
>    ![image](https://user-images.githubusercontent.com/23147096/164369863-1bc923ac-9d69-47cd-8cf3-59c7f454639f.png)

do you change something with the code? my training with ds_tcn config is not ok？

**Comment by mlxu995** [comment]
Thank you for your question.
We found that one parameter was missing in a previous code change, which may be the reason for the model's non-convergence.
In wekws/kws/utils/executor.py, line 48:
loss, acc = criterion(loss_type, logits, target, feats_lengths)
change to:
loss, acc = criterion(loss_type, logits, target, feats_lengths, min_duration)

**Comment by mlxu995** [comment]
Thank you for your question.
We found that one parameter was missing in a previous code change, which may be the reason for the model's non-convergence.
In wekws/kws/utils/executor.py, line 48:
loss, acc = criterion(loss_type, logits, target, feats_lengths)
change to:
loss, acc = criterion(loss_type, logits, target, feats_lengths, min_duration)

**Comment by mlxu995** [comment]
We have checked the result on xiaobu dataset using ds_tcn model.
the result seems good now ( after the aforementioned bug fixed ).

**Comment by mlxu995** [comment]
We have checked the result on xiaowen dataset using ds_tcn model.
the result seems good now ( after the aforementioned bug fixed ).

**Comment by sugarcase** [comment]
> 

Thank you very much. I will fix and try it later!
